mitosheet/mitosheet/step_performers/column_steps/split_text_to_columns.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# Copyright (c) Saga Inc.
+# Distributed under the terms of the GPL License.
+from copy import deepcopy
+import random
+import string
+from time import perf_counter
+from typing import Any, Dict, List, Optional, Set, Tuple
+import uuid
+
+import pandas as pd
+from mitosheet.code_chunks.code_chunk import CodeChunk
+from mitosheet.code_chunks.step_performers.column_steps.split_text_to_columns_code_chunk import SplitTextToColumnsCodeChunk
+from mitosheet.column_headers import try_make_new_header_valid_if_multi_index_headers
+from mitosheet.sheet_functions.types.utils import is_datetime_dtype
+from mitosheet.state import FORMAT_DEFAULT, State
+from mitosheet.step_performers.step_performer import StepPerformer
+from mitosheet.step_performers.utils import get_param
+from mitosheet.types import ColumnHeader, ColumnID
+
+def get_new_colum_header_unique_component() -> str:
+    return ''.join(random.choice(string.ascii_lowercase + string.digits) for _ in range(4))
+
+
+class SplitTextToColumnsStepPerformer(StepPerformer):
+    """""
+    A split_text_to_columns step, which allows you to separate 
+    a single column into multiple columns by specifying a delimiter
+    """
+
+    @classmethod
+    def step_version(cls) -> int:
+        return 1
+
+    @classmethod
+    def step_type(cls) -> str:
+        return 'split_text_to_columns'
+
+    @classmethod
+    def saturate(cls, prev_state: State, params: Dict[str, Any]) -> Dict[str, Any]:
+        return params
+
+    @classmethod
+    def execute(cls, prev_state: State, params: Dict[str, Any]) -> Tuple[State, Optional[Dict[str, Any]]]:
+        sheet_index: int = get_param(params, 'sheet_index')
+        column_id: ColumnID = get_param(params, 'column_id')
+        delimiters: List[str] = get_param(params, 'delimiters')
+        new_column_header_suffix: str = get_param(params, 'new_column_header_suffix')
+
+        column_header = prev_state.column_ids.get_column_header_by_id(sheet_index, column_id)
+        column_idx = prev_state.dfs[sheet_index].columns.tolist().index(column_header)
+            
+        # Create a new post state
+        post_state = prev_state.copy(deep_sheet_indexes=[sheet_index])
+        final_df = post_state.dfs[sheet_index]
+        delimiter_string = '|'.join(delimiters)
```

Comment:
What happens if we're splitting on a bar? TODO: test this.

Reply:
It seems to have the same issue as `.` do